servant/src/Servant/API/FileExtension.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE PolyKinds          #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections #-}
+
+module Servant.API.FileExtension
+    ( Ext(..)
+    , getExt
+    , parseExt
+    ) where
+
+import           Data.Typeable (Typeable)
+import           GHC.TypeLits  -- (Symbol)
+import           Web.HttpApiData
+import           Data.Text (Text, pack, stripSuffix)
+import           Data.Proxy
+import           Data.Maybe (catMaybes)
+import           Control.Monad ((>=>))
+
+-- | A wrapper around a time type which can be parsed/rendered to with `format',
```

Comment:
Whoops, yes, they we're developed at the tame time and I guess the two branches weren't kept as sterile as they should have been =)